Check in PyTorchModel

kathryn-baker opened the pull request.

roussel-ryan commented on lume_model/tests/pytorch/test_model.py:
```diff
@@ -0,0 +1,332 @@
+        try:
+            value.backward()
+            assert value.requires_grad
+        except AttributeError as exc:
```
I don't think you need the try/except statement here, the test will fail normally if value.backward() raises an error

kathryn-baker commented on lume_model/pytorch/__init__.py:
```diff
@@ -0,0 +1,334 @@
+logger = logging.getLogger(__name__)
+
+
+class PyTorchModel(BaseModel):
```
I've moved this as part of the latest changes

kathryn-baker commented:
A summary of what I've found out relating to the conda-build stage:
- the reason the build was being cancelled originally is because it couldn't solve the environment so I've removed the tensorflow and botorch requirements for the conda tests
- I've added an `importorskip` statement to the keras/torch tests so they won't run in the `conda` build stage but they should run in the `pip` stage, which will prevent the `conda` build from running if they fail so it's not a big loss in the test suite
- I've done a similar thing for any `FileNotFoundErrors`, any tests that require files in the `conda` build stage are skipped but should still be run in the `pip` stage
- I've kept the `setup.py install` call in the `build.sh` despite the warnings that this method is deprecated because using `pip` threw an error when I ran it locally and most other example recipes I've seen use the `setup.py` call
